packages/astro/test/0-css.test.js

```diff
@@ -0,0 +1,269 @@
+
+      // 2. check CSS
+      expect(bundledCSS).to.match(new RegExp(`.${moduleClass}[^{]*{font-family:fantasy}`));
+    });
+
+    it('.module.scss', async () => {
+      const $ = index$;
+      const el = $('#react-module-scss');
+      const classes = el.attr('class').split(' ');
+      const moduleClass = classes.find((name) => /^_title_[A-Za-z0-9-_]+/.test(name));
+
+      // 1. check HTML
+      expect(el.attr('class')).to.include(moduleClass);
+
+      // 2. check CSS
+      expect(bundledCSS).to.match(new RegExp(`.${moduleClass}[^{]*{font-family:fantasy}`));
+    });
+  });
+
+  describe('Vue', () => {
+    it('<style>', async () => {
+      const $ = index$;
+      const el = $('#vue-css');
+
+      // 1. check HTML
+      expect(el.attr('class')).to.include('vue-css');
+
+      // 2. check CSS
+      expect(bundledCSS).to.match(new RegExp(`.vue-css[^{]*{font-family:cursive`));
+    });
+
+    it('<style scoped>', async () => {
+      const $ = index$;
+      const el = $('#vue-scoped');
+
+      // find data-v-* attribute (how Vue CSS scoping works)
+      const { attribs } = el.get(0);
+      const scopeId = Object.keys(attribs).find((k) => k.startsWith('data-v-'));
+      expect(scopeId).to.be.ok;
+
+      // 1. check HTML
+      expect(el.attr('class')).to.include('vue-scoped');
+
+      // 2. check CSS
+      expect(bundledCSS).to.include(`.vue-scoped[${scopeId}]`);
+    });
+
+    it('<style module>', async () => {
+      const $ = index$;
+      const el = $('#vue-modules');
+      const classes = el.attr('class').split(' ');
+      const moduleClass = classes.find((name) => /^_title_[A-Za-z0-9-_]+/.test(name));
+
+      // 1. check HTML
+      expect(el.attr('class')).to.include(moduleClass);
+
+      // 2. check CSS
+      expect(bundledCSS).to.include(`${moduleClass}{`);
+    });
+
+    it('<style lang="sass">', async () => {
+      const $ = index$;
+      const el = $('#vue-sass');
+
+      // 1. check HTML
+      expect(el.attr('class')).to.include('vue-sass');
+
+      // 2. check CSS
+      expect(bundledCSS).to.match(new RegExp(`.vue-sass[^{]*{font-family:cursive`));
+    });
+
+    it('<style lang="scss">', async () => {
+      const $ = index$;
+      const el = $('#vue-scss');
+
+      // 1. check HTML
+      expect(el.attr('class')).to.include('vue-scss');
+
+      // 2. check CSS
+      expect(bundledCSS).to.match(new RegExp(`.vue-scss[^{]*{font-family:cursive`));
+    });
+  });
+
+  describe('Svelte', () => {
+    it('<style>', async () => {
+      const $ = index$;
+      const el = $('#svelte-css');
+      const classes = el.attr('class').split(' ');
+      const scopedClass = classes.find((name) => /^s-[A-Za-z0-9-]+/.test(name));
+
+      // 1. check HTML
+      expect(el.attr('class')).to.include('svelte-css');
+
+      // 2. check CSS
+      expect(bundledCSS).to.match(new RegExp(`.svelte-css.${scopedClass}[^{]*{font-family:"Comic Sans MS"`));
+    });
+
+    // TODO: fix Sass in Svelte
```

Comment:
Do we have an issue to track this? I want to get in the habit of creating issues for `TODO` comments.

Reply:
We do! https://github.com/snowpackjs/astro/issues/1976